rcl/test/rcl/test_time.cpp

```diff
@@ -196,101 +202,106 @@ TEST_F(CLASSNAME(TestTimeFixture, RMW_IMPLEMENTATION), test_rcl_init_for_time_so
   //   const int k_tolerance_ms = 1000;
   //   EXPECT_LE(llabs(now_diff), RCL_MS_TO_NS(k_tolerance_ms)) << "system_clock differs";
   // }
-  ret = rcl_time_point_fini(&a_time);
-  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string_safe();
-  ret = rcl_ros_time_source_fini(&source);
-  EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string_safe();
 }
 
-TEST(CLASSNAME(rcl_time, RMW_IMPLEMENTATION), time_source_validation) {
-  ASSERT_FALSE(rcl_time_source_valid(NULL));
-  rcl_time_source_t uninitialized;
+TEST(CLASSNAME(rcl_time, RMW_IMPLEMENTATION), clock_validation) {
+  ASSERT_FALSE(rcl_clock_valid(NULL));
+  rcl_clock_t uninitialized;
   // Not reliably detectable due to random values.
-  // ASSERT_FALSE(rcl_time_source_valid(&uninitialized));
+  // ASSERT_FALSE(rcl_clock_valid(&uninitialized));
+  rcl_allocator_t allocator = rcl_get_default_allocator();
   rcl_ret_t ret;
-  ret = rcl_ros_time_source_init(&uninitialized);
+  ret = rcl_ros_clock_init(&uninitialized, &allocator);
   EXPECT_EQ(ret, RCL_RET_OK) << rcl_get_error_string_safe();
 }
 
-TEST(CLASSNAME(rcl_time, RMW_IMPLEMENTATION), default_time_source_instanciation) {
-  rcl_time_source_t * ros_time_source = rcl_get_default_ros_time_source();
-  ASSERT_TRUE(rcl_time_source_valid(ros_time_source));
-  rcl_time_source_t * steady_time_source = rcl_get_default_steady_time_source();
-  ASSERT_TRUE(rcl_time_source_valid(steady_time_source));
-  rcl_time_source_t * system_time_source = rcl_get_default_system_time_source();
-  ASSERT_TRUE(rcl_time_source_valid(system_time_source));
+TEST(CLASSNAME(rcl_time, RMW_IMPLEMENTATION), default_clock_instanciation) {
+  rcl_clock_t * ros_clock =
+    reinterpret_cast<rcl_clock_t *>(calloc(1, sizeof(rcl_clock_t)));
```

Comment:
Does this need to be `calloc`ed for the test to pass? If someone created `rcl_clock_t` on the stack could random uninitialized values in the structure cause this test to fail?

Reply:
It does not need to be calloc specifically.